sdk/typescript/runtime/main.go

```diff
@@ -148,14 +210,65 @@ func (t *TypeScriptSdk) CodegenBase(ctx context.Context, modSource *ModuleSource
 			ContainerWithExecOpts{SkipEntrypoint: true}), nil
 }
 
-// Base returns a Node container with cache setup for yarn
-func (t *TypeScriptSdk) Base(version string) *Container {
-	if version == "" {
-		version = "21.3-alpine"
+// Base returns a Node or Bun container with cache setup for yarn or bun
+func (t *TypeScriptSdk) Base(runtime SupportedTSRuntime) (*Container, error) {
+	switch runtime {
+	case Bun:
+		return dag.Container().
+			From(bunImageRef).
+			WithMountedCache("/root/.bun/install/cache", dag.CacheVolume(fmt.Sprintf("mod-bun-cache-$s", bunVersion))).
+			WithoutEntrypoint(), nil
+	case Node:
+		return dag.Container().
+			From(nodeImageRef).
+			WithMountedCache("/root/.npm", dag.CacheVolume(fmt.Sprintf("mod-npm-cache-%s", nodeVersion))).
+			WithoutEntrypoint(), nil
+	default:
+		return nil, fmt.Errorf("unknown runtime: %v", runtime)
+	}
+}
+
+// DetectRuntime returns the runtime(bun or node) detected for the user's module
+// If a runtime is specfied inside the package.json, it will be used.
+// If a package-lock.json is present, node will be used.
+// If a bun.lockb is present, bun will be used.
+// If none of the above is present, node will be used.
+func (t *TypeScriptSdk) DetectRuntime(ctx context.Context, modSource *ModuleSource, subPath string) (SupportedTSRuntime, error) {
+	// Try to detect runtime from package.json
+	source := modSource.ContextDirectory()
+	// read contents of package.json
+	runtime, err := func() (SupportedTSRuntime, error) {
+		json, err := source.File(path.Join(subPath, "package.json")).Contents(ctx)
+		if err != nil {
+			return "", fmt.Errorf("could not read package.json: %w", err)
+		}
+
+		value := gjson.Get(json, "dagger.runtime").String()
+		if value == "" {
+			return "", fmt.Errorf("no runtime specified in package.json")
+		}
```

Comment:
It looks like setting "runtime" in "package.json" is now required, even for "node", but these errors are actually ignored.

Readability could be better if in `value == ""` you go look for the lock files, and then do the switch case in scope, with a value that you know isn't empty (`switch SupportedTSRuntime(value)`).